### DISCUSSION TITLE
Obtain the vector component from the GirdFunction in HDiv space

### DISCUSSION THREAD

**Discussion by Liu0813** [mfem/mfem#1916] [Q&A] [answered]
Hi, everyone,
I want to modify the miniapp Joule and solve a 2d axisymmetric model. However, I encountered some problems about the heat conduction equation. In this miniapp, the heat conduction equation is solved using mixed finite element method, i.e. the temperature flux is defined in HDiv space, and the temperature is defined in L2 space. A simple method of solving axisymmetric model is to add a source term `Fr/r` in 2d heat conduction equation, that is
![image](https://user-images.githubusercontent.com/30075822/100977055-6368e380-357b-11eb-9376-644540528f8c.png)

Thus, I have to obtain the `Fr` component. But for HDiv space, the function `ProjectCoefficient` doesn't work. So I hope to know how to project the r-component of temperature flux to a new GridFunction defined in L2 space.

**Comment by tzanio** [comment, marked as answer]
If I understand correctly, you'd like to compute say the `x`-component of an H(div) field `F` in 2D (in your notation `r=x`).

You can do that by projecting the continuous coefficient `(1,0)` to the RT space and then performing dots product with `F` in each degree of freedom.

Does this make sense?
Tzanio

**Comment by Liu0813** [comment]
Thanks for your reply. The method you provided is helpful.

**Comment by tzanio** [comment]
Converted to Discussion...
